dev/env.go

```diff
@@ -0,0 +1,240 @@
+		c.Logger = logr.Discard()
+	}
+	if c.NewCluster == nil {
+		c.NewCluster = NewCluster
+	}
+	if c.ClusterOptions == nil {
+		c.ClusterOptions = append(c.ClusterOptions, WithLogger(c.Logger))
+	}
+}
+
+type EnvironmentOption interface {
+	ApplyToEnvironmentConfig(c *EnvironmentConfig)
+}
+
+type WithClusterInitializers []ClusterInitializer
+
+func (i WithClusterInitializers) ApplyToEnvironmentConfig(c *EnvironmentConfig) {
+	c.ClusterInitializers = append(c.ClusterInitializers, i...)
+}
+
+type WithContainerRuntime string
+
+func (cr WithContainerRuntime) ApplyToEnvironmentConfig(c *EnvironmentConfig) {
+	c.ContainerRuntime = string(cr)
+}
+
+type WithNewClusterFunc NewClusterFunc
+
+func (f WithNewClusterFunc) ApplyToEnvironmentConfig(c *EnvironmentConfig) {
+	c.NewCluster = NewClusterFunc(f)
+}
+
+type WithClusterOptions []ClusterOption
+
+func (opts WithClusterOptions) ApplyToEnvironmentConfig(c *EnvironmentConfig) {
+	c.ClusterOptions = opts
+}
+
+type NewClusterFunc func(kubeconfigPath string, opts ...ClusterOption) (*Cluster, error)
+
+type ClusterInitializer interface {
+	Init(ctx context.Context, cluster *Cluster) error
+}
+
+// Load objects from given folder paths and applies them into the cluster.
+type ClusterLoadObjectsFromFolders []string
+
+func (l ClusterLoadObjectsFromFolders) Init(
+	ctx context.Context, cluster *Cluster) error {
+	return cluster.CreateAndWaitFromFolders(ctx, l)
+}
+
+// Load objects from given file paths and applies them into the cluster.
+type ClusterLoadObjectsFromFiles []string
+
+func (l ClusterLoadObjectsFromFiles) Init(
+	ctx context.Context, cluster *Cluster) error {
+	return cluster.CreateAndWaitFromFiles(ctx, l)
+}
+
+type ClusterLoadObjectsFromHttp []string
+
+func (l ClusterLoadObjectsFromHttp) Init(
+	ctx context.Context, cluster *Cluster) error {
+	return cluster.CreateAndWaitFromHttp(ctx, l)
+}
+
+type ClusterHelmInstall struct {
+	RepoName, RepoURL, PackageName, Namespace, ReleaseName string
+	SetVars                                                []string
+}
+
+func (l ClusterHelmInstall) Init(
+	ctx context.Context, cluster *Cluster) error {
+	return cluster.HelmInstall(
+		ctx, cluster,
+		l.RepoName, l.RepoURL, l.PackageName, l.ReleaseName, l.Namespace, l.SetVars)
+}
+
+// Environment represents a development environment.
+type Environment struct {
+	Name string
+	// Working directory of the environment.
+	// Temporary files/kubeconfig etc. will be stored here.
+	WorkDir string
+	Cluster *Cluster
+	config  EnvironmentConfig
+}
+
+// Creates a new development environment.
+func NewEnvironment(name, workDir string, opts ...EnvironmentOption) *Environment {
+	env := &Environment{
+		Name:    name,
+		WorkDir: workDir,
+	}
+	for _, opt := range opts {
+		opt.ApplyToEnvironmentConfig(&env.config)
+	}
+	env.config.Default()
+	return env
+}
+
+// Initializes the environment and prepares it for use.
+func (env *Environment) Init(ctx context.Context) error {
+	kindConfig := `kind: Cluster
```

Comment:
nit: Would refactor to helper func that constructs the kind config file.

Reply:
nit: This function is a zillion lines long it is yelling for a refactor :sweat_smile: